galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+        # Don't need these anymore, so free memory
+        del self._phaseFT, self.psi, self.screen
+
+        # To handle wind, we will interpolate the LookupTable2D with an offset origin.
+        self.origin = np.r_[0.0, 0.0]
+
+    def __str__(self):
+        return "galsim.FrozenAtmosphericScreen(altitude=%s)" % self.altitude
+
+    def __repr__(self):
+        outstr = ("galsim.FrozenAtmosphericScreen(%r, %r, altitude=%r, time_step=%r, " +
+                  "r0_500=%r, L0_inv=%r, vx=%r, vy=%r, rng=%r)")
+        return outstr % (self.screen_size, self.screen_scale, self.altitude, self.time_step,
+                         self.r0_500, self.L0_inv, self.vx, self.vy, self.rng)
+
+    def __eq__(self, other):
+        return (self.screen_size == other.screen_size and
+                self.screen_scale == other.screen_scale and
+                self.altitude == other.altitude and
+                self.vx == other.vx and
+                self.vy == other.vy and
+                self.tab2d == other.tab2d and
+                np.array_equal(self.origin, other.origin))
+
+    def __ne__(self, other):
+        return not self == other
+
+    def advance(self):
+        """Advance phase screen realization by self.time_step."""
+        # If wind blows right, then origin moves left, so use minus sign.
+        self.origin -= (self.vx*self.time_step, self.vy*self.time_step)
+
+    def advance_by(self, dt):
+        """Advance phase screen by specified amount of time.
+
+        @param dt  Amount of time in seconds by which to update the screen.
+        @returns   The actual amount of time updated, which is equal to the input argument in this
+                   case.
+        """
+        self.origin -= (self.vx*dt, self.vy*dt)
+        return dt
+
+    def path_difference(self, nx, scale, theta_x=0.0*galsim.degrees, theta_y=0.0*galsim.degrees):
+        """ Compute effective pathlength differences due to phase screen.
+
+        @param nx       Size of output array
+        @param scale    Scale of output array pixels in meters
+        @param theta_x  Field angle corresponding to center of output array.
+        @param theta_y  Ditto.
+        @returns   Array of pathlength differences in nanometers.  Multiply by 2pi/wavelength to get
+                   array of phase differences.
+        """
+        xmin = self.origin[0] + 1000*self.altitude*theta_x.tan() - 0.5*scale*(nx-1)
+        # xmin = self.origin[0] + 1000*self.altitude*theta_x.tan() - 0.5*scale*nx
+        xmax = xmin + scale*(nx-1)
+        ymin = self.origin[1] + 1000*self.altitude*theta_y.tan() - 0.5*scale*(nx-1)
+        # ymin = self.origin[1] + 1000*self.altitude*theta_y.tan() - 0.5*scale*nx
+        ymax = ymin + scale*(nx-1)
+        return self.tab2d.eval_grid(xmin, xmax, nx, ymin, ymax, nx)
+
+    def reset(self):
+        """Reset phase screen back to time=0."""
+        self.origin = np.r_[0.0, 0.0]
```

Comment:
>This is kind of a non-standard usage for `r_`.

Ah right.  That's my lazy way of making arrays interactively, but I guess I let it leak into actual .py on accident.  Changed.